arcgis-ios-sdk-samples/Layers/ArcGIS vector tiled layer (custom style)/VectorStylesViewController.swift

```diff
@@ -16,36 +16,34 @@
 
 import UIKit
 
-protocol VectorStylesViewControllerDelegate: AnyObject {
+protocol VectorStylesVCDelegate: AnyObject {
     func vectorStylesViewController(_ vectorStylesViewController: VectorStylesViewController, didSelectItemWithID itemID: String)
 }
 
 class VectorStylesViewController: UITableViewController {
+    /// The item IDs of the custom styles.
     var itemIDs: [String] = []
+    /// The item ID of the selected style.
     var selectedItemID: String?
     
-    weak var delegate: VectorStylesViewControllerDelegate?
+    weak var delegate: VectorStylesVCDelegate?
     
-    override func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        if let selectedItemID = selectedItemID,
-            let row = itemIDs.firstIndex(of: selectedItemID) {
-            let indexPath = IndexPath(row: row, section: 0)
-            tableView.selectRow(at: indexPath, animated: false, scrollPosition: .none)
-            let cell = tableView.cellForRow(at: indexPath)
-            cell?.accessoryType = .checkmark
-        }
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = super.tableView(tableView, cellForRowAt: indexPath)
+        // Indicate the displayed item.
+        cell.accessoryType = itemIDs[indexPath.row] == selectedItemID ? .checkmark : .none
+        return cell
     }
     
-    // MARK: - UITableViewDelegate
-
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         let itemID = itemIDs[indexPath.row]
         delegate?.vectorStylesViewController(self, didSelectItemWithID: itemID)
+        // Indicate which cell has been selected.
         tableView.cellForRow(at: indexPath)?.accessoryType = .checkmark
-    }
-    
-    override func tableView(_ tableView: UITableView, didDeselectRowAt indexPath: IndexPath) {
-        tableView.cellForRow(at: indexPath)?.accessoryType = .none
+        
+        if let previousItemID = selectedItemID, let previousRow = itemIDs.firstIndex(of: previousItemID) {
+            tableView.cellForRow(at: IndexPath(row: previousRow, section: 0))?.accessoryType = .none
+        }
```

Comment:
I'd suggest swap the order of these 2 lines. Otherwise, when tapping on a cell for multiple times, it won't display the checkmark.
Or consider `guard let` early in the function to abort further changes if `selectedItemID` is not changed.
![multiple](https://user-images.githubusercontent.com/9660181/99699519-e13ed200-2a46-11eb-9c6d-f22a33a176a8.gif)

Reply:
Nice catch! I reorganized. Please take a look!